src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -282,7 +282,7 @@ export default function SavedObjectFactory(esAdmin, kbnIndex, Promise, Private,
           if (_.get(err, 'origError.status') === 409) {
             const confirmMessage = `Are you sure you want to overwrite ${this.title}?`;
 
-            return safeConfirm(confirmMessage)
+            return confirmModalPromise(confirmMessage, { confirmButtonText: 'Overwrite' })
```

Comment:
Since it's only being used in two places, I think we should just create and return the promises locally at each call site, and just use `confirmModal`.

If you wanted to dig deeper and see if the promises are even necessary, then I think that's a good optional second step.

Either way, unless there's a compelling reason to use `confirmModalPromise` over `confirmModal`, then I think we're adding unnecessary complexity by keeping both. If there are clear use-cases where `confirmModalPromise` should be used over `confirmModal`, then I think we need to change the name to reflect that use-case (instead of simply reflecting an implementation detail).